### ISSUE TITLE
run build.cmd failed 

### ISSUE THREAD

**Issue by SmGG** [aspnet/Mvc#8243]
#### Description of the problem:

run build.cmd failed ,  
Access to the path 'Internal.AspNetCore.Sdk.dll' is denied  ,
The SDK 'Internal.AspNetCore.Sdk/2.2.0-preview1-20180731.1' specified could not be found.

Can anybody help me with this!!

info down here

>$ build.cmd /t:Restore
Using KoreBuild 2.2.0-preview1-20180731.1
WARNING: dotnet found on the system PATH is 'C:\Program Files\dotnet\dotnet.exe' but KoreBuild will use
'C:\Users\Smile_Gum_Gum\.dotnet\x64\dotnet.exe'.
WARNING: Adding 'C:\Users\Smile_Gum_Gum\.dotnet\x64' to system PATH permanently may be required for applications
like Visual Studio or VS Code to work correctly.
Adding C:\Users\Smile_Gum_Gum\.dotnet\x64 to PATH
.NET Core SDK 2.1.302 is already installed. Skipping installation.
>>> dotnet.exe msbuild -restore -t:noop -v:m C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\scripts/../modules/BundledPackages/BundledPackageRestorer.csproj
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.
C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\modules\BundledPackages\BundledPackageRestorer.csproj : error : Access to the path 'Internal.AspNetCore.Sdk.dll' is denied.
C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\modules\BundledPackages\BundledPackageRestorer.csproj : error MSB4236: The SDK 'Internal.AspNetCore.Sdk/2.2.0-preview1-20180731.1' specified could not be found.
dotnet.exe failed with exit code: 1
所在位置 C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\scripts\common.psm1:11 字符
: 9
+         throw "$cmdName failed with exit code: $exitCode"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (dotnet.exe failed with exit code: 1:String) [], RuntimeException
    + FullyQualifiedErrorId : dotnet.exe failed with exit code: 1

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @SmGG.
@ryanbrandenburg can you please look into this?

**Comment by ryanbrandenburg** [comment]
Sounds like there's a process that has a handle on 'Internal.AspNetCore.Sdk.dll', so it can't be used. Easiest way to see if that's true is to either use Process Explorer to search for the handle, or just restart the computer so that ALL processes end. Another possibility is that the file actually doesn't exist, but one of the landmarks KoreBuild uses to decide on the state does. This could happen because the process was interrupted in the middle of expanding the folder. You can test that out by deleting your `C:\Users\Smile_Gum_Gum\.dotnet` folder and letting KoreBuild re-create it.

I think one of those should get you sorted. If not knowing a little more about the situation that lead to this state would help. Have you run any of our repo builds on this machine before? How long ago if so? Did it succeed then? That kind of thing.

**Comment by pranavkm** [comment]
`C:\Users\Smile_Gum_Gum.dotnet\`

The path looks wonky. Should be `C:\Users\Smile_Gum_Gum\.dotnet\`.

**Comment by ryanbrandenburg** [comment]
@pranavkm I though that too but if you look at the actual text (by clicking edit) you'll see that the \ is there. It must be a Github escaping problem of some sort.

**Comment by SmGG** [comment]
@ryanbrandenburg  I tried again this morning and got a different error,  then I did as you say remove the directory C:\Users\Smile_Gum_Gum\.dotnet  , run build.cmd again still the same error, the output like below

>$ build.cmd
Downloading KoreBuild 2.2.0-preview1-20180731.1
Using KoreBuild 2.2.0-preview1-20180731.1
WARNING: dotnet found on the system PATH is 'C:\Program Files\dotnet\dotnet.exe' but KoreBuild will use
'C:\Users\Smile_Gum_Gum\.dotnet\x64\dotnet.exe'.
WARNING: Adding 'C:\Users\Smile_Gum_Gum\.dotnet\x64' to system PATH permanently may be required for applications
like Visual Studio or VS Code to work correctly.
Adding C:\Users\Smile_Gum_Gum\.dotnet\x64 to PATH
dotnet-install: Downloading link: https://dotnetcli.blob.core.windows.net/dotnet/Sdk/2.1.302/dotnet-sdk-2.1.302-win-x64.zip
dotnet-install: Extracting zip from https://dotnetcli.blob.core.windows.net/dotnet/Sdk/2.1.302/dotnet-sdk-2.1.302-win-x64.zip
dotnet-install: Adding to current process PATH: "C:\Users\Smile_Gum_Gum\.dotnet\x64\". Note: This change will not be visible if PowerShell was run as a child process.
dotnet-install: Installation finished
>>> dotnet.exe msbuild @E:\PersonalProject\dotnetMvcCore\Mvc\artifacts\logs\msbuild.rsp @E:\PersonalProject\dotnetMvcCore\Mvc\artifacts\logs\msbuild.logger.rsp

>Welcome to .NET Core!
>---------------------
>Learn more about .NET Core: https://aka.ms/dotnet-docs
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs

>Telemetry
>---------
>The .NET Core tools collect usage data in order to help us improve your experience. The data is anonymous and doesn't include command-line arguments. The data is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

>Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

>Configuring...
>--------------
>A command is running to populate your local package cache to improve restore speed and enable offline access. This command takes up to one minute to complete and only runs once.
Decompressing 100% 7028 ms
Expanding 100% 15002 ms

>ASP.NET Core
>------------
>Successfully installed the ASP.NET Core HTTPS Development Certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only). For establishing trust on other platforms refer to the platform specific documentation.
For more information on configuring HTTPS see https://go.microsoft.com/fwlink/?linkid=848054.
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

>  Installing .NET Core Runtime (x64) 2.1.0
  dotnet-install output:
  VERBOSE: dotnet-install: Get-CLIArchitecture-From-Architecture -Architecture "x64"
  VERBOSE: dotnet-install: Get-Specific-Version-From-Version -AzureFeed "https://dotnetcli.azureedge.net/dotnet" -Channel "LTS" -Version "2.1.0"
  VERBOSE: dotnet-install: Get-Download-Link -AzureFeed "https://dotnetcli.azureedge.net/dotnet" -SpecificVersion "2.1.0" -CLIArchitecture "x64"
  VERBOSE: dotnet-install: Constructed primary payload URL: https://dotnetcli.azureedge.net/dotnet/Runtime/2.1.0/dotnet-runtime-2.1.0-win-x64.zip
  VERBOSE: dotnet-install: Get-LegacyDownload-Link -AzureFeed "https://dotnetcli.azureedge.net/dotnet" -SpecificVersion "2.1.0" -CLIArchitecture "x64"
  VERBOSE: dotnet-install: Constructed legacy payload URL: https://dotnetcli.azureedge.net/dotnet/Runtime/2.1.0/dotnet-win-x64.2.1.0.zip
  VERBOSE: dotnet-install: Resolve-Installation-Path -InstallDir "C:\Users\Smile_Gum_Gum\.dotnet\x64"
  VERBOSE: dotnet-install: InstallRoot: C:\Users\Smile_Gum_Gum\.dotnet\x64
  VERBOSE: dotnet-install: Is-Dotnet-Package-Installed -InstallRoot "C:\Users\Smile_Gum_Gum\.dotnet\x64" -RelativePathToPackage "shared\Microsoft.NETCore.App" -SpecificVersion "2.1.0"
  VERBOSE: dotnet-install: Is-Dotnet-Package-Installed: Path to a package: C:\Users\Smile_Gum_Gum\.dotnet\x64\shared\Microsoft.NETCore.App\2.1.0
  VERBOSE: Perform operation 'Enumerate CimInstances' with following parameters, ''namespaceName' = root\cimv2,'className' = win32_logicaldisk'.
  VERBOSE: 完成操作“Enumerate CimInstances”。
  VERBOSE: dotnet-install: Zip path: C:\Users\Smile_Gum_Gum\AppData\Local\Temp\q52rkioz.tf4
  dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/2.1.0/dotnet-runtime-2.1.0-win-x64.zip
C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\modules\KoreBuild.Tasks\module.targets(47,5): error : dotnet-install of .NET Core Runtime (x64) 2.1.0 timed out after 240 seconds. [C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\KoreBuild.proj]

>Build FAILED.

>C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\modules\KoreBuild.Tasks\module.targets(47,5): error : dotnet-install of .NET Core Runtime (x64) 2.1.0 timed out after 240 seconds. [C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\KoreBuild.proj]
    0 Warning(s)
    1 Error(s)

>Time Elapsed 00:04:01.20
dotnet.exe failed with exit code: 1
所在位置 C:\Users\Smile_Gum_Gum\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\scripts\common.psm1:11 字符: 9
>+         throw "$cmdName failed with exit code: $exitCode"
>+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>    + CategoryInfo          : OperationStopped: (dotnet.exe failed with exit code: 1:String) [], >RuntimeException
>    + FullyQualifiedErrorId : dotnet.exe failed with exit code: 1


when I start to run build.cmd, it all goes well, but when it comes to this, it seems like blocking
>  Installing .NET Core Runtime (x64) 2.1.0
and after 240 seconds, it goes wrong, is that something like a network problem?
Please help me, thanks!

Images here:

Blocking here:
![rz j5j223uxe0n8b0mu 49](https://user-images.githubusercontent.com/24875099/43937379-dcf7a79a-9c8f-11e8-8b7a-bd9200609a90.png)

Then Error:
![mcbqf_ zs t 2 yaaacm u](https://user-images.githubusercontent.com/24875099/43937382-e11591d4-9c8f-11e8-978c-34ece66c40b6.png)

**Comment by mkArtakMSFT** [comment]
@SmGG have you tried to download the file manually from the pointed out URL: https://dotnetcli.azureedge.net/dotnet/Runtime/2.2.0-preview1-26618-02/dotnet-runtime-2.2.0-preview1-26618-02-win-x64.zip

**Comment by mkArtakMSFT** [comment]
@ryanbrandenburg, how can the timeout value be extended?

**Comment by ryanbrandenburg** [comment]
@SmGG, if you modify `~\.dotnet\buildtools\korebuild\2.2.0-preview1-17091\modules\KoreBuild.Tasks\module.targets` to replace:
```
<InstallDotNet
      Assets="@(DotNetCoreSdk);@(DotNetCoreRuntime);@(AspNetCoreRuntime)"
      DotNetHome="$(DOTNET_HOME)"
      InstallScript="$(_DotNetInstall)"/>
```
with
```
<InstallDotNet
      Assets="@(DotNetCoreSdk);@(DotNetCoreRuntime);@(AspNetCoreRuntime)"
      DotNetHome="$(DOTNET_HOME)"
      InstallScript="$(_DotNetInstall)"
      TimeoutSeconds=<SomeLargerNumber>
      />
```

that should do it. How big that number is depends on how fast your internet is.

**Comment by SmGG** [comment]
@ryanbrandenburg  so it is a network problem? 
 I run build.cmd again just now,
First, I still got the timeout problem,
Then, I try to download the dotnet-runtime-2.2.0-preview1-26618-02-win-x64.zip manually,
At the same time, I change the config, add the TimeoutSeconds=3600, 
I run it again, found out it should be written like this TimeoutSeconds="3600",
I run it another time, and I don't know how and why it works
Now I got the issue #8252,
Thanks a lot, @ryanbrandenburg, @mkArtakMSFT,  seem like my problem get fixed, and now, let me wait for you guys to fix #8252

**Comment by ryanbrandenburg** [comment]
Glad we could help. Yeah, most likely you had a slow network, or a spotty connection between your network and where the runtime is hosted.
